File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0){
            return 0;
        }
        sort(nums.begin(),nums.end());
        int maxCount = 1 ;
        int count = 1;

        for(int i = 1 ; i < nums.size() ;i++){
            if(nums[i] == nums[i-1]){
                continue;
            }
            if(nums[i] == nums[i-1]+1){
                count++;
            }else{
                count = 1;
            }
            maxCount = max(count,maxCount);
        }

        return maxCount;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        if(nums.length == 0)return 0;

        HashSet<Integer> set = new HashSet<>();

        for(int i : nums){
            set.add(i);
        }

        int max = 1 ;

        for(int i : nums){

            if(!set.contains(i-1)){
                int count = 0;
                int n = i ;
                while(set.contains(n)){
                    count++;
                    set.remove(n);
                    n = n + 1;
                }
                max = Math.max(max,count);
            }
            
        }

        return max ; 

    }
}